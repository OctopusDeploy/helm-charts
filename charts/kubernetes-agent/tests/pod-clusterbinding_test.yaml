suite: "pod permissions"
templates:
- templates/pod-clusterbinding.yaml
tests:
- it: "is not created when target namespaces is not empty"
  set:
    scriptPods:
      serviceAccount:
        targetNamespaces: ["ns-1"]
  asserts:
  - hasDocuments:
      count: 0

- it: "is not created when useNamespacedRoles is true"
  set:
    scriptPods:
      serviceAccount:
        useNamespacedRoles: true
  asserts:
  - hasDocuments:
      count: 0

- it: "should match snapshot"
  asserts:
  - matchSnapshot: {}
    
- it: "worker should not attach cluster role binding to serviceAccount"
  set:
    agent.worker.enabled: true
  asserts:
    - hasDocuments:
        count: 0