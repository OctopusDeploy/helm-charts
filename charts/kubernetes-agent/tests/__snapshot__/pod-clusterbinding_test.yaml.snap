should match snapshot:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: octopus-agent-scripts-RELEASE-NAME-binding
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: octopus-agent-scripts-RELEASE-NAME-role
    subjects:
      - kind: ServiceAccount
        name: octopus-agent-scripts
        namespace: NAMESPACE
